### ISSUE TITLE
Unexpected error when calling `unpack128` on field in array of fields.

### ISSUE THREAD

**Issue by mark-moir** [Zokrates/ZoKrates#1108]
## Description

A trivial program that calls `unpack128` on the first field of an array of two fields causes an error:
```
called `Result::unwrap()` on an `Err` value: UnconstrainedVariable (zokrates_core/src/proof_system/bellman/mod.rs:191:47)
This is unexpected, please submit a full bug report at https://github.com/Zokrates/ZoKrates/issues
```

This error occurs with the default G16 scheme, but not with G17, which performs setup successfully.

Also, if I replace the array of two fields with a single field, and pass that to `unpack128`, the error does not occur.

And if I remove the `private` from the function signature, the error does not occur.

## Environment

- Compiler version: 0.7.11  (656923c)
- Operating system: Darwin 19.6.0 (Mac OS Catalina 10.15.7)

## Steps to Reproduce
```
> echo $ZOKRATES_STDLIB
<<redacted>>/zokrates-0.7.11/stdlib

> which $Z
<<redacted>>/zokrates-0.7.11/zokrates
```
Create minimal.zok with the following contents:

```zokrates
import "utils/pack/u32/unpack128" as unpack128

def main(private field[2] flds) -> u32[4]:
    return unpack128(flds[0])
```

```
> $Z compile -i minimal.zok -o minimal.out -s minimal-abi.json
Compiling minimal.zok

Compiled code written to 'minimal.out'

> $Z setup -i minimal.out -p minimal-proving.key -v minimal-verification.keyPerforming setup...
WARNING: You are using the G16 scheme which is subject to malleability. See zokrates.github.io/toolbox/proving_schemes.html#g16-malleability for implications.
called `Result::unwrap()` on an `Err` value: UnconstrainedVariable (zokrates_core/src/proof_system/bellman/mod.rs:191:47)
This is unexpected, please submit a full bug report at https://github.com/Zokrates/ZoKrates/issues

> $Z setup --proving-scheme gm17 --backend ark -i minimal.out -p minimal-proving.key -v minimal-verification.key
Performing setup...
Verification key written to 'minimal-verification.key'
Proving key written to 'minimal-proving.key'
Setup completed
```

**Comment by dark64** [comment]
Hi @mark-moir, this is expected, all private inputs must be used. Although this error could be handled more gracefully.

**Comment by mark-moir** [comment]
@dark64, thanks, good to know.  Yes, that error message could be improved :smile:.

**Comment by Schaeff** [comment]
This was fixed in #1120
